app/main.py

```diff
@@ -1,2 +1,93 @@
+from typing import Any, Union
+
+
 class Dictionary:
-    pass
+    # визначаємо основні аргументи таблиці: розмір початкової табли, коеф. загруженості, розмір наповнення
+    def __init__(
+            self,
+            capasity: int = 8,
+            load_factor: float = 2 / 3,
+            size: int = 0,
+    ) -> None:
+        self.load_factor = load_factor
+        self.capasity = capasity
+        self.size = size
+        # створюємо дефолтну пусту таблицю
+        self.table = [[] for _ in range(self.capasity)]
+        self.overload = round(self.capasity * self.load_factor)
+
+    # в цьому методі вираховуємо індекс елемента, за формулою, щоб не поіторювати цю формулу в інших методах
+    def hash_func(self, key: Any) -> int:
+        return hash(key) % self.capasity
+
+    # пишемо основний функціонал, як вираховується місце елемента в таблиці,
+    # як відбувається заповнення таблиці
+    # проходить перевірка чи немає колізії(комірка зайнята іншим елементом)
+    # чи заміна елемента якщо мають одинакові ключі
+    def __setitem__(self, key: Any, value: Any) -> None:
+        hash_key = hash(key)
+        index_key = self.hash_func(key)
+        self.resize(key, value)
+        if self.size == 0:
+            self.table[index_key].append((key, value, hash_key))
+            self.size += 1
+        else:
+            for elem in range(len(self.table[index_key])):
+                if self.table[index_key][elem][0] == key:
+                    self.table[index_key][elem][1] = (key, value, hash_key)
+
+            self.table[index_key + 1].append((key, value, hash_key))
+            self.size += 1
+
+    # прописуємо розширення таблиці при заповненні на 2/3
+    def resize(self, key: Any, value: Any) -> None:
+        if self.size >= self.overload:
+            self.capasity *= 2
+
+            self.table = [[] for _ in range(self.capasity)]
+            self.__setitem__(key, value)
```

Comment:
```suggestion
            self[key] = value
```

Reply:
I do't understand
Where I need to add it?